### ISSUE TITLE
api.openweathermap.org removed snow? 

### ISSUE THREAD

**Issue by michbeck100** [josecastroleon/pimatic-openweather#28]
Since a few days i'm getting errors when starting pimatic with the open weather plugin:

```
21:26:57.776 [pimatic-openweather] error: Object.keys called on non-object
21:26:58.801 [pimatic-openweather] debug: TypeError: Object.keys called on non-object
21:26:58.801 [pimatic-openweather] debug:>    at Function.keys (native)
21:26:58.801 [pimatic-openweather] debug:>    at PromiseRetryer.run.then._ref (/home/pi/pimatic-app/node_modules/pimatic-openweather/openweather.coffee:129:61)
21:26:58.801 [pimatic-openweather] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/util.js:26:23)
21:26:58.801 [pimatic-openweather] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/promise.js:507:31)
21:26:58.801 [pimatic-openweather] debug:>    at Promise._settlePromiseAt (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/promise.js:581:18)
21:26:58.801 [pimatic-openweather] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/promise.js:697:14)
21:26:58.801 [pimatic-openweather] debug:>    at Async._drainQueue (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/async.js:123:16)
21:26:58.801 [pimatic-openweather] debug:>    at Async._drainQueues (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/async.js:133:10)
21:26:58.801 [pimatic-openweather] debug:>    at Async.drainQueues (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/async.js:15:14)
21:26:58.801 [pimatic-openweather] debug:>    at process._tickCallback (node.js:415:13)
```

The line in open weather.coffee tries to extract the values for snow. I've looked at the openweather api and it seems the api doesn't have any values for snow (anymore?), at least for my location (which is Oldenburg, Germany).

This is my config: 

```
{
      "id": "wetter",
      "name": "Wetter",
      "class": "OpenWeatherDevice",
      "location": "Oldenburg, Germany",
      "units": "metric",
      "lang": "de",
      "timeout": 300000
}
```

**Comment by mwittig** [comment]
Note, as part of the location you should provide the 2-letter ISO 3166 country code rather than providing the country name."  

~~It seems like th weather data for Oldenburg is incomplete. I get complete results on the forecast, but "snow" is missing for current weather. The API specification http://openweathermap.org/current states "If you do not see some of the parameters in your API respond it means that these weather phenomena are just not happened for the time of measurement for the city or location chosen. Only really measured or calculated data is displayed in API respond." So, my understanding is this is normal behaviour, but of course, it should not raise exceptions.~~

Having investigated this I found a bug in the code handling the weather updates on the OpenWeatherDevice. I'll provide a fix shortly.

**Comment by mwittig** [comment]
Fixed with release 0.8.18
